src/HucaresServer.Storage.UnitTests/MissingLicensePlateTests.cs

```diff
@@ -0,0 +1,213 @@
+using Shouldly;
+
+namespace HucaresServer.Storage.UnitTests
+{
+    [TestClass]
+    public class MissingLicensePlateTests
+    {
+        [TestMethod]
+        public void InsertMissingPlate_WhenPlateNumberIsValid_ShouldSucceedAndReturnExpected()
+        {
+            //Arrange
+            var fakeIQueryable = new List<MissingLicensePlate>().AsQueryable();
+            var fakeDbSet = A.Fake<DbSet<MissingLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingLicensePlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            //Act
+            var expectedPlate = "JBA:555";
+            var dateInput = "Jan 1, 2018";
+            DateTime expectedDate = DateTime.Parse(dateInput);
+            var result = missingLicensePlateHelper.InsertPlateRecord(expectedPlate, expectedDate);
+
+            //Assert
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => fakeDbSet.Add(result))
+                .MustHaveHappened();
+
+            A.CallTo(() => fakeHucaresContext.SaveChanges())
+                .MustHaveHappenedOnceExactly();
+
+            result.PlateNumber.ShouldBe(expectedPlate);
+            result.SearchStartDateTime.ShouldBe(expectedDate);
+        }
+
+        [TestMethod]
+        public void InsertMissingPlate_WhenPlateNumberIsNotValid_ShouldThrowAnError()
+        {
+            //Arrange
+            var fakeDbSet = A.Fake<DbSet<MissingLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingPlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            var dateInput = "Jan 1, 2018";
+            DateTime expectedDate = DateTime.Parse(dateInput);
+            
+            //Act & Assert
+            Assert.ThrowsException<UriFormatException>(() => missingPlateHelper.InsertPlateRecord("5555555", expectedDate));
+
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustNotHaveHappened();
+        }
+
+        [TestMethod]
+        public void GetAllPlates_WhenDbIsEmpty_ShouldThrowEmptyDB()
+        {
+            //Arrange
+            var missingPlateObj = new MissingLicensePlate();
+            var fakeIQueryable = new List<MissingLicensePlate>()
```

Comment:
Name is "when db is empty", however you are faking out values. This test fails because it expects no records.